src/ProxyClient/Request/RequestQueue.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\ProxyClient\Request;
+
+use FOS\HttpCache\Exception\InvalidUrlException;
+use Http\Discovery\UriFactoryDiscovery;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+
+/**
+ * A queue of requests to be sent to the HTTP caching server
+ */
+class RequestQueue implements \Countable
```

Comment:
I would play with the word 'batch' somehow. Also, if it is connected to invalidation, why isn't that word in the name? (like InvalidationQueue?)